includes/REST_API/Publisher_Logos_Controller.php

```diff
@@ -0,0 +1,428 @@
+	 *
+	 * @param WP_REST_Request $request Full details about the request.
+	 * @return WP_REST_Response|WP_Error Response object on success, or WP_Error object on failure.
+	 */
+	public function delete_item( $request ) {
+		$post = $this->get_publisher_logo( $request['id'] );
+
+		if ( is_wp_error( $post ) ) {
+			return $post;
+		}
+
+		$publisher_logos = $this->filter_publisher_logos( (array) get_option( Settings::SETTING_NAME_PUBLISHER_LOGOS, [] ) );
+		$publisher_logos = array_values( array_diff( $publisher_logos, [ $post->ID ] ) );
+
+		$active_publisher_logo_id = absint( get_option( Settings::SETTING_NAME_ACTIVE_PUBLISHER_LOGO ) );
+
+		if ( $post->ID === $active_publisher_logo_id || ! in_array( $active_publisher_logo_id, $publisher_logos, true ) ) {
+			// Mark the first available publisher logo as the new default.
+			update_option( Settings::SETTING_NAME_ACTIVE_PUBLISHER_LOGO, ! empty( $publisher_logos[0] ) ? $publisher_logos[0] : 0 );
+		}
+
+		update_option( Settings::SETTING_NAME_PUBLISHER_LOGOS, $publisher_logos );
+
+		return $this->prepare_item_for_response( $post, $request );
+	}
+
+	/**
+	 * Updates a publisher logo in the collection.
+	 *
+	 * Can only be used to make it the "active" one.
+	 *
+	 * @since 1.12.0
+	 *
+	 * @param WP_REST_Request $request Full details about the request.
+	 * @return WP_REST_Response|WP_Error Response object on success, or WP_Error object on failure.
+	 */
+	public function update_item( $request ) {
+		$post = $this->get_publisher_logo( $request['id'] );
+
+		if ( is_wp_error( $post ) ) {
+			return $post;
+		}
+
+		if ( $request['active'] ) {
+			update_option( Settings::SETTING_NAME_ACTIVE_PUBLISHER_LOGO, $post->ID );
+		}
+
+		return $this->prepare_item_for_response( $post, $request );
+	}
+
+	/**
+	 * Get an existing publisher logo's post object, if valid.
+	 *
+	 * @since 1.12.0
+	 *
+	 * @param int $id Supplied ID.
+	 * @return WP_Post|WP_Error Post object if ID is valid, WP_Error otherwise.
+	 */
+	protected function get_publisher_logo( $id ) {
+		$publisher_logos = $this->filter_publisher_logos( (array) get_option( Settings::SETTING_NAME_PUBLISHER_LOGOS, [] ) );
+
+		$post = get_post( $id );
+
+		if ( ! $post || ! in_array( $post->ID, $publisher_logos, true ) ) {
+			return new WP_Error(
+				'rest_invalid_id',
+				__( 'Invalid ID', 'web-stories' ),
+				[ 'status' => 400 ]
+			);
+		}
+
+		return $post;
+	}
+
+	/**
+	 * Prepares a single publisher logo output for response.
+	 *
+	 * @since 1.12.0
+	 *
+	 * @param WP_Post         $post    Post object.
+	 * @param WP_REST_Request $request Request object.
+	 * @return WP_REST_Response Response object.
+	 */
+	public function prepare_item_for_response( $post, $request ): WP_REST_Response {
+		$fields = $this->get_fields_for_response( $request );
+
+		// Base fields for every post.
+		$data = [];
+
+		if ( rest_is_field_included( 'id', $fields ) ) {
+			$data['id'] = $post->ID;
+		}
+
+		if ( rest_is_field_included( 'title', $fields ) ) {
```

Comment:
Instead of embedding title and url, does make sense to embed the attachment object, see https://github.com/google/web-stories-wp/pull/9101/files#r714074117

Reply:
Let's discuss in the other thread